File: auth-service/src/main/java/com/mealmap/authservice/core/dto/UserRegisterDto.java
package com.mealmap.authservice.core.dto;

import com.mealmap.authservice.core.enums.UserRole;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class UserRegisterDto {
    String phoneNumber;

    String email;

    String password;

    String firstName;

    String lastName;

    UserRole role;
}

File: frontend/src/api/auth/dto/UserRegisterDto.ts
import { Role } from "../enums/Role";

export interface UserRegisterDto {
  phoneNumber: string; // Must match PHONE_BELARUS_FORMAT regex
  email: string; // Valid email format
  password: string; // Between 4 and 50 characters
  firstName: string; // Max 50 characters
  lastName: string; // Max 50 characters
  organizationId: number;
  role: Role;
}
